.github/workflows/docker-sandbox-cache.yml

```diff
@@ -20,104 +25,103 @@ jobs:
         # Remove old layers first to help with disk usage
+        echo "Before Prune:"
+        docker images
         docker system prune --force
-        pull_docker_cache ${ORG}/${CACHE_IMAGE}:${MAIN_TAG} ${ORG}/${CACHE_IMAGE}:non-existing-image-tag
+        echo "After Prune:"
+        docker images
+        ci-helper pull_docker_cache ${im_stage2} ${im_stage2_fallback} 
+        echo "After Pull:"
+        docker images
 
     - name: Build Sandbox Docker (stage2)
       run: |
-        source .github/workflows/ci-helpers.sh
         # now build second stage making sure first stage is from cache
         docker build \
-          --cache-from ${ORG}/${CACHE_IMAGE}:${BUILDER_TAG} \
-          --cache-from ${ORG}/${CACHE_IMAGE}:${MAIN_TAG} \
-          --tag        ${ORG}/${IMAGE}:latest \
+          --build-arg BUILD_INFO="${build_info}" \
+          --cache-from ${im_stage1} \
+          --cache-from ${im_stage2} \
+          --tag        ${im_stage2} \
           ./docker/
 
-    - name: DockerHub Push Cache (stage2)
-      if: github.event_name == 'push'
+    - name: DockerHub Push Image (stage2)
+      if: steps.cfg.outputs.push_image == 'yes'
       run: |
-        source .github/workflows/ci-helpers.sh
-        docker tag ${ORG}/${IMAGE}:latest ${ORG}/${CACHE_IMAGE}:${MAIN_TAG}
-        docker push ${ORG}/${CACHE_IMAGE}:${MAIN_TAG}
+        docker push ${im_stage2}
 
     - name: Build Sandbox Docker (sudo)
       run: |
-        source .github/workflows/ci-helpers.sh
-        # build SUDO version
         docker build \
           --build-arg WITH_SUDO=yes \
-          --cache-from ${ORG}/${CACHE_IMAGE}:${BUILDER_TAG} \
-          --cache-from ${ORG}/${IMAGE}:latest \
-          --tag        ${ORG}/${IMAGE}:sudo-latest \
+          --build-arg BUILD_INFO="${build_info} (sudo)" \
+          --cache-from ${im_stage1} \
+          --cache-from ${im_stage2} \
+          --tag        ${im_sudo} \
           ./docker/
 
-    # If on master, push latest and tags including sudo versions
-    - name: Get the Git tag (master branch)
-      if: github.ref == 'refs/heads/master'
-      id: get_version
+    - name: DockerHub Push Image (sudo)
+      if: steps.cfg.outputs.push_image == 'yes'
       run: |
-        git fetch --prune --unshallow
-        echo ::set-output name=VERSION::$(git describe --tags)
+        docker push ${im_sudo}
 
-    - name: DockerHub Push (latest and tag)
-      if: github.ref == 'refs/heads/master'
-      run: |
-        source .github/workflows/ci-helpers.sh
-        docker push ${ORG}/${IMAGE}:latest
-        docker tag ${ORG}/${IMAGE}:latest ${ORG}/${IMAGE}:${{ steps.get_version.outputs.VERSION }}
-        docker push ${ORG}/${IMAGE}:${{ steps.get_version.outputs.VERSION }}
 
-    - name: DockerHub Push (sudo and sudo-tag)
-      if: github.ref == 'refs/heads/master'
+    - name: Push Extra tags (master branch only)
+      if: github.ref == 'refs/heads/master' && steps.cfg.outputs.push_image == 'yes'
```

Comment:
Do you think this'll handle the tags properly?

My logic before was:
- if master, push `latest` and `sudo-latest`
- if on a release, push `release-tag` and -`sudo-release-tag`

That way means that the release doesn't push to latest. And you explicitly get the release tags pushed.

- 

-

Reply:
I don't push release to latest either.

Reply:
Ah, got it. I see the logic now.

I like the PR not pushing too, that's tidy.